apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -1785,6 +1843,115 @@ static void libc_stdio_fileno_tc(void)
 }
 
 /**
+* @testcase         libc_stdio_remove
+* @brief            Deletes the file whose name is specified in filename.
+* @scenario         Open file
+* @apicovered       freopen
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_remove(void)
+{
+	char *filename = VFS_FILE_PATH;
+	int ret = NULL;
+
+	ret = remove(filename);
+	TC_ASSERT_EQ("remove", ret, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_setbuf
+* @brief            Set stream buffer
+* @scenario         Open file
+B
+* @apicovered       setbuf, setvbuf
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_setbuf(void)
+{
+	FILE *fp;
+	char buffer[64];
+	char *filename = VFS_FILE_PATH;
+	int ret = NULL;
+
+	/* setbuf_test: DEFAULT buffering */
+
+	fp = fopen(filename, "w");
+	TC_ASSERT_NEQ("setbuf", fp, NULL);
+
+	/* setbuf_test: NO buffering */
+
+	setbuf(fp, NULL);
+	TC_ASSERT_LEQ("setbuf", ret, 0);
+
+	/* setbuf_test: pre-allocated buffer */
+
+	setbuf(fp, buffer);
+	TC_ASSERT_LEQ("setbuf", ret, 0);
+	setbuf(fp, NULL);
+	TC_ASSERT_LEQ("setbuf", ret, 0);
+
+	fclose(fp);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_setvbuf
+* @brief            Change stream buffering
+* @scenario         Open file
+* @apicovered       setbuf
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_setvbuf(void)
+{
+	FILE *fp;
+	char buffer[64];
+	char *filename = VFS_FILE_PATH;
+	int ret = NULL;
+
+	/* setvbuf_test: DEFAULT buffering */
+
+	fp = fopen(filename, "w");
+
+	TC_ASSERT_NEQ("setvbuf", fp, NULL);
+
+	/* setvbuf_test: NO buffering */
+
+	ret = setvbuf(fp, NULL, _IONBF, 0);
+	TC_ASSERT_LEQ("setvbuf", ret, 0);
```

Comment:
No, if TC fails here, it will return immediately and `fclose(fp)` will never be reached.  check `TC_ASSERT_XX` macros.